### ISSUE TITLE
setup IBLI data processing server

### ISSUE THREAD

**Issue by niryariv** [Gizra/IBLI#18]

**Comment by alanorth** [comment]
I plan to install an Ubuntu 12.04 LTS server so we can start working on:
- Playing with Matlab on the command-line-only environment
- External IP address
- Holes poked in ILRI firewall for external access

Question: we still aren't sure how much disk space we'll need, eh? I typically allocate ~30GB for my VMs, but with this data (even if it's not long-lived), I'm not sure if that's enough.

**Comment by niryariv** [comment]
I understood from Anton the IDL code does look at NDVI history, so we'll need to have some room for that. What the max we could allocate?

**Comment by alanorth** [comment]
Server is installed, Ubuntu 12.04.4.  I've allocated 128GB disk space for the virtual machine.  I'll keep you posted as I get it set up and start testing stuff.

**Comment by woodardjoshua** [comment]
Alan, did the initial matlab run work for you?

On Thu, Apr 10, 2014 at 2:26 AM, Alan Orth notifications@github.com wrote:

> Server is installed, Ubuntu 12.04.4. I've allocated 128GB disk space for
> the virtual machine. I'll keep you posted as I get it set up and start
> testing stuff.
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/Gizra/IBLI/issues/18#issuecomment-40046866
> .

## 

---

Joshua D. Woodard, PhD
Assistant Professor of Agribusiness and Finance
Zaitz Family Sesquicentenntial Faculty Fellow
David R. Atkinson Center for a Sustainable Future Faculty Fellow
Dyson School of Applied Economics and Management
Cornell University
Phone: (217) 979-1251 Fax: (815) 328-2252
Email: woodardjoshua@gmail.com, jdw277@cornell.edu
Webpage: http://dyson.cornell.edu/people/profiles/woodard.php

---

Confidentiality Notice: This message, including any attachments, is for the
sole use of the intended recipient(s) and may contain confidential and
privileged information. Any unauthorized review, use, disclosure, or
distribution is prohibited.  This email is considered a draft of material
that could be published at a later date, and should not be forwarded,
cited, or quoted without permission, and is subject to revision. If you are
not the intended recipient, please contact the sender by reply e-mail and
destroy all copies of the original message. We do not accept legal
responsibility or otherwise for the contents of this message.

**Comment by alanorth** [comment]
Installing Matlab now... the headless install (without X11) is a pain in the ass, so I've installed some X11 libs and am installing with `ssh -X`... I'll keep you posted.

**Comment by alanorth** [comment]
Ok, MatLab is running on the server now... :thumbsup:

Installation notes in this gist: https://gist.github.com/alanorth/baf2b6d79759c405d95e

Next I will try to run the code @apurbashee gave me.

**Comment by woodardjoshua** [comment]
Make sure it is most recent version. I sent it to Nir and believe it is on
github. Thanks!

Josh

**Comment by alanorth** [comment]
@woodardjoshua Ok, cloning the Gizra/IBLI repo right now...

**Comment by alanorth** [comment]
Ok, I got it running on the server but I get an error about a missing `fminunc` function. Output here: https://gist.githubusercontent.com/alanorth/4e537b6d63ed980ff082/raw/1d00212502aa2376ce4de36cd72d9b22a71aef39/gistfile1.txt

**Comment by woodardjoshua** [comment]
It means that you do not the have the optimization toolbox loaded. Do you
have the optimization toolbox?

On Fri, Apr 11, 2014 at 9:30 AM, Alan Orth notifications@github.com wrote:

> Ok, I got it running on the server but I get an error about a missing
> fminunc function. Output here:
> https://gist.githubusercontent.com/alanorth/4e537b6d63ed980ff082/raw/1d00212502aa2376ce4de36cd72d9b22a71aef39/gistfile1.txt
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/Gizra/IBLI/issues/18#issuecomment-40203176
> .

## 

---

Joshua D. Woodard, PhD
Assistant Professor of Agribusiness and Finance
Zaitz Family Sesquicentenntial Faculty Fellow
David R. Atkinson Center for a Sustainable Future Faculty Fellow
Dyson School of Applied Economics and Management
Cornell University
Phone: (217) 979-1251 Fax: (815) 328-2252
Email: woodardjoshua@gmail.com, jdw277@cornell.edu
Webpage: http://dyson.cornell.edu/people/profiles/woodard.php

---

Confidentiality Notice: This message, including any attachments, is for the
sole use of the intended recipient(s) and may contain confidential and
privileged information. Any unauthorized review, use, disclosure, or
distribution is prohibited.  This email is considered a draft of material
that could be published at a later date, and should not be forwarded,
cited, or quoted without permission, and is subject to revision. If you are
not the intended recipient, please contact the sender by reply e-mail and
destroy all copies of the original message. We do not accept legal
responsibility or otherwise for the contents of this message.

**Comment by alanorth** [comment]
@woodardjoshua That's a good question.  I'm trying to figure out how to see the packages I have installed... in Matlab...

I see it on the Mathworks website, but it looks like a separate license?

http://www.mathworks.com/products/optimization/

**Comment by alanorth** [comment]
Hmmm, looks like I don't have the toolbox.

``` console
>> ver
-------------------------------------------------------------------------------------------
MATLAB Version: 8.3.0.532 (R2014a)
MATLAB License Number: xxxxxx
Operating System: Linux 3.2.0-60-generic #91-Ubuntu SMP Wed Feb 19 03:54:44 UTC 2014 x86_64
Java Version: Java is not enabled
-------------------------------------------------------------------------------------------
MATLAB                                                Version 8.3        (R2014a)
```

**Comment by woodardjoshua** [comment]
Hmmm...well, you can either buy it, or if we don't want to expend the funds
then we need to do some sort of work around. Since we are going to
unconditional rates for the next offering, we can do that, but I will need
to send you a new set of files.

My recommendation is you save the money and let me send you the new set of
files, just means that we will be committed to unconditional rates at least
for now. Note that these files that I sent are the ones Apurba asked for,
but I am actually working on this stuff all day today so can send them over
in a bit if that works.

Josh

On Fri, Apr 11, 2014 at 9:43 AM, Alan Orth notifications@github.com wrote:

> Hmmm, looks like I don't have the toolbox.
> 
> > > ver-------------------------------------------------------------------------------------------MATLAB Version: 8.3.0.532 (R2014a)MATLAB License Number: 935005Operating System: Linux 3.2.0-60-generic #91-Ubuntu SMP Wed Feb 19 03:54:44 UTC 2014 x86_64Java Version: Java is not enabled-------------------------------------------------------------------------------------------MATLAB                                                Version 8.3        (R2014a)
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/Gizra/IBLI/issues/18#issuecomment-40204484
> .

## 

---

Joshua D. Woodard, PhD
Assistant Professor of Agribusiness and Finance
Zaitz Family Sesquicentenntial Faculty Fellow
David R. Atkinson Center for a Sustainable Future Faculty Fellow
Dyson School of Applied Economics and Management
Cornell University
Phone: (217) 979-1251 Fax: (815) 328-2252
Email: woodardjoshua@gmail.com, jdw277@cornell.edu
Webpage: http://dyson.cornell.edu/people/profiles/woodard.php

---

Confidentiality Notice: This message, including any attachments, is for the
sole use of the intended recipient(s) and may contain confidential and
privileged information. Any unauthorized review, use, disclosure, or
distribution is prohibited.  This email is considered a draft of material
that could be published at a later date, and should not be forwarded,
cited, or quoted without permission, and is subject to revision. If you are
not the intended recipient, please contact the sender by reply e-mail and
destroy all copies of the original message. We do not accept legal
responsibility or otherwise for the contents of this message.

**Comment by alanorth** [comment]
@woodardjoshua Ah, ok. Proprietary software is fun; licenses for everything! ;)

Well, maybe IBLI has a few hundred bucks laying around and we can get the Optimization Toolbox addon for Matlab.  What do you think, @bnedavies?

**Comment by apurbashee** [comment]
@alanorth @woodardjoshua this was what I was trying to decide before buying.. but no problem we will buy this toolbox. I do not think we need any other than optimization toolbox. what do you think @woodardjoshua ?

**Comment by woodardjoshua** [comment]
I don't think you need anything but that one as far as I can tell.

On Fri, Apr 11, 2014 at 10:18 AM, Apurba Shee notifications@github.comwrote:

> @alanorth https://github.com/alanorth @woodardjoshuahttps://github.com/woodardjoshuathis was what I was trying to decide before buying.. but no problem we will
> buy this toolbox. I do not think we need any other than optimization
> toolbox. what do you think @woodardjoshuahttps://github.com/woodardjoshua?
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/Gizra/IBLI/issues/18#issuecomment-40208216
> .

## 

---

Joshua D. Woodard, PhD
Assistant Professor of Agribusiness and Finance
Zaitz Family Sesquicentenntial Faculty Fellow
David R. Atkinson Center for a Sustainable Future Faculty Fellow
Dyson School of Applied Economics and Management
Cornell University
Phone: (217) 979-1251 Fax: (815) 328-2252
Email: woodardjoshua@gmail.com, jdw277@cornell.edu
Webpage: http://dyson.cornell.edu/people/profiles/woodard.php

---

Confidentiality Notice: This message, including any attachments, is for the
sole use of the intended recipient(s) and may contain confidential and
privileged information. Any unauthorized review, use, disclosure, or
distribution is prohibited.  This email is considered a draft of material
that could be published at a later date, and should not be forwarded,
cited, or quoted without permission, and is subject to revision. If you are
not the intended recipient, please contact the sender by reply e-mail and
destroy all copies of the original message. We do not accept legal
responsibility or otherwise for the contents of this message.

**Comment by woodardjoshua** [comment]
..but by the way Apurba, like I mentioned, since we are going to
unconditional rates you will not need it then. You might want it though as
we move forward there will probably be things we will use it for. Up to you
whether you want to get it now or when we get to that point. I can send the
new set of files in the meantime if you prefer that. Let me know.

On Fri, Apr 11, 2014 at 10:20 AM, Joshua Woodard woodardjoshua@gmail.comwrote:

> I don't think you need anything but that one as far as I can tell.
> 
> On Fri, Apr 11, 2014 at 10:18 AM, Apurba Shee notifications@github.comwrote:
> 
> > @alanorth https://github.com/alanorth @woodardjoshuahttps://github.com/woodardjoshuathis was what I was trying to decide before buying.. but no problem we will
> > buy this toolbox. I do not think we need any other than optimization
> > toolbox. what do you think @woodardjoshuahttps://github.com/woodardjoshua?
> > 
> > ## 
> > 
> > Reply to this email directly or view it on GitHubhttps://github.com/Gizra/IBLI/issues/18#issuecomment-40208216
> > .
> 
> ## 
> 
> ---
> 
> Joshua D. Woodard, PhD
> Assistant Professor of Agribusiness and Finance
> Zaitz Family Sesquicentenntial Faculty Fellow
>  David R. Atkinson Center for a Sustainable Future Faculty Fellow
> Dyson School of Applied Economics and Management
> Cornell University
> Phone: (217) 979-1251 Fax: (815) 328-2252
> Email: woodardjoshua@gmail.com, jdw277@cornell.edu
> Webpage: http://dyson.cornell.edu/people/profiles/woodard.php
> 
> ---
> 
> Confidentiality Notice: This message, including any attachments, is for
> the sole use of the intended recipient(s) and may contain confidential and
> privileged information. Any unauthorized review, use, disclosure, or
> distribution is prohibited.  This email is considered a draft of material
> that could be published at a later date, and should not be forwarded,
> cited, or quoted without permission, and is subject to revision. If you are
> not the intended recipient, please contact the sender by reply e-mail and
> destroy all copies of the original message. We do not accept legal
> responsibility or otherwise for the contents of this message.

## 

---

Joshua D. Woodard, PhD
Assistant Professor of Agribusiness and Finance
Zaitz Family Sesquicentenntial Faculty Fellow
David R. Atkinson Center for a Sustainable Future Faculty Fellow
Dyson School of Applied Economics and Management
Cornell University
Phone: (217) 979-1251 Fax: (815) 328-2252
Email: woodardjoshua@gmail.com, jdw277@cornell.edu
Webpage: http://dyson.cornell.edu/people/profiles/woodard.php

---

Confidentiality Notice: This message, including any attachments, is for the
sole use of the intended recipient(s) and may contain confidential and
privileged information. Any unauthorized review, use, disclosure, or
distribution is prohibited.  This email is considered a draft of material
that could be published at a later date, and should not be forwarded,
cited, or quoted without permission, and is subject to revision. If you are
not the intended recipient, please contact the sender by reply e-mail and
destroy all copies of the original message. We do not accept legal
responsibility or otherwise for the contents of this message.

**Comment by woodardjoshua** [comment]
Alan and Nir,

Please see the following link for the new set of matlab files, which
includes the live imports and should not require optimization toolbox.
https://www.dropbox.com/sh/5eu1ihna3dgwa04/_24MtcIH7l

This is LRLD, but once you get this one working, I will drop in the SRSD
for the same.

Note that there is a help file in that folder that explains exactly how to
run the code and where the data need to be, and what it exports. Note that
in the specified data folders, these are exactly the data that Anton gave
me, so they need to be formatted the same for all the imports to work.

Anton, please see the folder "z-scoring_first_CalibratedSeries" for how you
sent it before and how it is being imported in the code. See finalNDVI
folder for where/how the final post and pre values should be for
calculating indemnities and final index values.

This is unconditional, so you should be able to move forward with this code
without the optimization toolbox. This was in flux prior, but Apurba
confirmed this with me Wed.

So you can scrap the previous set of matlab code now.

Instructions to generate premium rates and final indexes and inemdnities:

1) Open genLRLDrate in matlab
2) Ensure LRLDbasedata.mat is in same path, and that base NDVI data is in
folder called z-scoring_first_CalibratedSeries in same folder.
3) Ensure that final end season NDVI data is in folder called finalNDVI in
same folder.
4) Run code (F5)

Final rates as well as Final Index and Indemnity/loss rate values are
processed in this function and will export those csv's.

Will export LRLDratefinal10.csv and LRLDratefinal15.csv, the final rates,
as well as final indemnities (%loss) and final index values as
LRLDratefinalIndem10.csv, LRLDratefinalIndem15.csv, and LRLDfinalIndex.csv
for all 108 divisions.

Same steps for SRSD.

Let me know if you have any questions and hopefully this pretty much does
it. We will want to be able to swap out .m files easily, but if it is set
up like this, then I should be able to keep the imports consistent.

Thanks,

Josh

On Fri, Apr 11, 2014 at 12:10 PM, Joshua Woodard woodardjoshua@gmail.comwrote:

> ..but by the way Apurba, like I mentioned, since we are going to
> unconditional rates you will not need it then. You might want it though as
> we move forward there will probably be things we will use it for. Up to you
> whether you want to get it now or when we get to that point. I can send the
> new set of files in the meantime if you prefer that. Let me know.
> 
> On Fri, Apr 11, 2014 at 10:20 AM, Joshua Woodard woodardjoshua@gmail.comwrote:
> 
> > I don't think you need anything but that one as far as I can tell.
> > 
> > On Fri, Apr 11, 2014 at 10:18 AM, Apurba Shee notifications@github.comwrote:
> > 
> > > @alanorth https://github.com/alanorth @woodardjoshuahttps://github.com/woodardjoshuathis was what I was trying to decide before buying.. but no problem we will
> > > buy this toolbox. I do not think we need any other than optimization
> > > toolbox. what do you think @woodardjoshuahttps://github.com/woodardjoshua?
> > > 
> > > ## 
> > > 
> > > Reply to this email directly or view it on GitHubhttps://github.com/Gizra/IBLI/issues/18#issuecomment-40208216
> > > .
> > 
> > ## 
> > 
> > ---
> > 
> > Joshua D. Woodard, PhD
> > Assistant Professor of Agribusiness and Finance
> > Zaitz Family Sesquicentenntial Faculty Fellow
> >  David R. Atkinson Center for a Sustainable Future Faculty Fellow
> > Dyson School of Applied Economics and Management
> > Cornell University
> > Phone: (217) 979-1251 Fax: (815) 328-2252
> > Email: woodardjoshua@gmail.com, jdw277@cornell.edu
> > Webpage: http://dyson.cornell.edu/people/profiles/woodard.php
> > 
> > ---
> > 
> > Confidentiality Notice: This message, including any attachments, is for
> > the sole use of the intended recipient(s) and may contain confidential and
> > privileged information. Any unauthorized review, use, disclosure, or
> > distribution is prohibited.  This email is considered a draft of material
> > that could be published at a later date, and should not be forwarded,
> > cited, or quoted without permission, and is subject to revision. If you are
> > not the intended recipient, please contact the sender by reply e-mail and
> > destroy all copies of the original message. We do not accept legal
> > responsibility or otherwise for the contents of this message.
> 
> ## 
> 
> ---
> 
> Joshua D. Woodard, PhD
> Assistant Professor of Agribusiness and Finance
> Zaitz Family Sesquicentenntial Faculty Fellow
>  David R. Atkinson Center for a Sustainable Future Faculty Fellow
> Dyson School of Applied Economics and Management
> Cornell University
> Phone: (217) 979-1251 Fax: (815) 328-2252
> Email: woodardjoshua@gmail.com, jdw277@cornell.edu
> Webpage: http://dyson.cornell.edu/people/profiles/woodard.php
> 
> ---
> 
> Confidentiality Notice: This message, including any attachments, is for
> the sole use of the intended recipient(s) and may contain confidential and
> privileged information. Any unauthorized review, use, disclosure, or
> distribution is prohibited.  This email is considered a draft of material
> that could be published at a later date, and should not be forwarded,
> cited, or quoted without permission, and is subject to revision. If you are
> not the intended recipient, please contact the sender by reply e-mail and
> destroy all copies of the original message. We do not accept legal
> responsibility or otherwise for the contents of this message.

## 

---

Joshua D. Woodard, PhD
Assistant Professor of Agribusiness and Finance
Zaitz Family Sesquicentenntial Faculty Fellow
David R. Atkinson Center for a Sustainable Future Faculty Fellow
Dyson School of Applied Economics and Management
Cornell University
Phone: (217) 979-1251 Fax: (815) 328-2252
Email: woodardjoshua@gmail.com, jdw277@cornell.edu
Webpage: http://dyson.cornell.edu/people/profiles/woodard.php

---

Confidentiality Notice: This message, including any attachments, is for the
sole use of the intended recipient(s) and may contain confidential and
privileged information. Any unauthorized review, use, disclosure, or
distribution is prohibited.  This email is considered a draft of material
that could be published at a later date, and should not be forwarded,
cited, or quoted without permission, and is subject to revision. If you are
not the intended recipient, please contact the sender by reply e-mail and
destroy all copies of the original message. We do not accept legal
responsibility or otherwise for the contents of this message.

**Comment by alanorth** [comment]
Ok, downloaded the new code.  I will try running it tomorrow (Tuesday) and let you know how it goes!

**Comment by alanorth** [comment]
I made some edits in pull request #22 and both `genLRLDrate.m` and `genSRSDrate.m` are running now in Linux, but getting getting errors towards the end.

Here's the log for a run of SRSD: https://gist.githubusercontent.com/alanorth/31ccb055d8e4fde86778/raw/c039839a5f0c6a293a66856ca26dd450150f52c1/gistfile1.txt

**Comment by woodardjoshua** [comment]
Alan,

The error says your import is not working. Ensure that the file you are
trying to import is in the correct place, has the file correct name, and
that the contents of the file are set up and formatted the exact same as
the original files that Anton provided (I included a test file with what I
sent).

One thing that is confusing on your output though, it says the error is at
line 29 on an import routine but when I look the code that is not what is
at line 29. Why is that?

Also can you clarify why you had to "linuxize" the code?  Apologies I have
never used matlab in linux.

I made some edits in pull request
#22https://github.com/Gizra/IBLI/pull/22and both
genLRLDrate.m and genSRSDrate.m are running now in Linux, but getting
getting errors towards the end.

Here's the log for a run of SRSD:
https://gist.githubusercontent.com/alanorth/31ccb055d8e4fde86778/raw/c039839a5f0c6a293a66856ca26dd450150f52c1/gistfile1.txt

—
Reply to this email directly or view it on
GitHubhttps://github.com/Gizra/IBLI/issues/18#issuecomment-40459413
.

**Comment by alanorth** [comment]
@woodardjoshua well the only really Linux-izing that was **mandatory** was using forward slashes instead of backslashes in paths.  The second commit converts the line endings to use Unix line endings, which is pretty important if we're editing the files in Linux / git.  The third just cleans up a bunch of trailing white space which is kinda a nitpick of mine. :)

I will look at the imports later.  Maybe I missed a path...

**Comment by woodardjoshua** [comment]
Alan, thanks much.

All your imports look good except the last one. That file for that import
is for generating an evaluation of the current value of the index. The
format of that file should be essentially a 108 x 3 array, with the first
column being the generic location ID (1-108), the second being the current
season value of preseason NDVI (so in the case of the SRSD file the
previous LRLD season index value) and the third column should be the
in-season/post NDVI (i.e., for SRSD file, the current SRSD value).

I attach that file. Can you check to see if yours looks like that and is in
the same location \IBLIMatlab\finalNDVI\finalNDVI_SRSD.csv

If it is not that, then there are several possibilities of what it could be
on your end...a couple links maybe to help.

http://stackoverflow.com/questions/10606373/what-causes-an-invalid-file-identifier-in-matlab

http://www.mathworks.com/matlabcentral/newsreader/view_thread/166338

What is strange is that the other imports were a bit more involved and
those worked, but the one that was very simple straightforward and did not.
So I assume it is something small. If it keeps creating problems, since it
is such a simple import, you might just try to quickly rebuild it on your
end. Let me know if a call would help.

Forget my question about the line 29 thing.

On Tue, Apr 15, 2014 at 6:29 AM, Alan Orth notifications@github.com wrote:

> @woodardjoshua https://github.com/woodardjoshua well the only really
> Linux-izing that was _mandatory_ was using forward slashes instead of
> backslashes in paths. The second commit converts the line endings to use
> Unix line endings, which is pretty important if we're editing the files in
> Linux / git. The third just cleans up a bunch of trailing white space which
> is kinda a nitpick of mine. :)
> 
> I will look at the imports later. Maybe I missed a path...
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/Gizra/IBLI/issues/18#issuecomment-40466959
> .

## 

---

Joshua D. Woodard, PhD
Assistant Professor of Agribusiness and Finance
Zaitz Family Sesquicentenntial Faculty Fellow
David R. Atkinson Center for a Sustainable Future Faculty Fellow
Dyson School of Applied Economics and Management
Cornell University
Phone: (217) 979-1251 Fax: (815) 328-2252
Email: woodardjoshua@gmail.com, jdw277@cornell.edu
Webpage: http://dyson.cornell.edu/people/profiles/woodard.php

---

Confidentiality Notice: This message, including any attachments, is for the
sole use of the intended recipient(s) and may contain confidential and
privileged information. Any unauthorized review, use, disclosure, or
distribution is prohibited.  This email is considered a draft of material
that could be published at a later date, and should not be forwarded,
cited, or quoted without permission, and is subject to revision. If you are
not the intended recipient, please contact the sender by reply e-mail and
destroy all copies of the original message. We do not accept legal
responsibility or otherwise for the contents of this message.

**Comment by alanorth** [comment]
Ok!  Found another Windows-style path that needed to be fixed (#23).  Now the code runs to completion and produces, for example (SRSD):

```
SRSDfinalIndem10.csv
SRSDfinalIndem15.csv
SRSDfinalIndex.csv
SRSDratefinal10.csv
SRSDratefinal15.csv
```

**Comment by woodardjoshua** [comment]
Yes great! Is it clear from the read me file I included as to what those
outputs are?
